### PR TITLE
feat(alerts): Move blocking member list request

### PR DESCRIPTION
**Pull request by scttcper**
Move blocking member list request to the sidebar where its used. This request is used to display the member's name in the sidebar. eg - `Send a notification to Dave` if its a specific member.

speeds up the page by not blocking the page load on the users request. Could be 1-2 seconds for some orgs

The red line shows where the page was waiting for users request to finish before loading 2 more requests.
image

### CODE REVIEW

File: static/app/views/alerts/rules/issue/details/ruleDetails.spec.jsx [hunk 1]
```diff
@@ -161,20 +162,11 @@ describe('AlertRuleDetails', () => {
...
       ],
     });
-    MockApiClient.mockResponses.splice(
-      MockApiClient.mockResponses.findIndex(
-        response =>
-          response.url ===
-          `/projects/${organization.slug}/${project.slug}/rules/${rule.id}/`
-      ),
-      1
-    );
```
Comment (scttcper) [inline static/app/views/alerts/rules/issue/details/ruleDetails.spec.jsx:177]: just cleaning up some unnecessary stuff

File: static/app/views/alerts/rules/issue/details/sidebar.tsx [hunk 2]
```diff
@@ -9,18 +9,26 @@ import TimeSince from 'sentry/components/timeSince';
...
-  memberList: Member[];
+  projectSlug: string;
   rule: IssueAlertRule;
   teams: Team[];
 };
 
-function Conditions({rule, teams, memberList}: Props) {
+function Conditions({rule, teams, projectSlug}: Props) {
+  const organization = useOrganization();
+  const {data: memberList} = useApiQuery<Member[]>(
```
Comment (malwilley) [inline static/app/views/alerts/rules/issue/details/sidebar.tsx:27]: Not necessary to do now of course but it seems like we can/should replace other member list calls with this in the future

File: static/app/views/alerts/rules/issue/details/sidebar.tsx [hunk 3]
```diff
@@ -80,7 +88,11 @@ function Conditions({rule, teams, memberList}: Props) {
               rule.actions.map((action, idx) => {
                 return (
                   <ConditionsBadge key={idx}>
-                    <TextAction action={action} memberList={memberList} teams={teams} />
+                    <TextAction
+                      action={action}
+                      memberList={memberList ?? []}
```
Comment (malwilley) [inline static/app/views/alerts/rules/issue/details/sidebar.tsx:93]: What does this look like when a user action is here? Does it handle things nicely while the request is in flight?

Comment (scttcper) [reply static/app/views/alerts/rules/issue/details/sidebar.tsx:93]: added a better fallback to Unknown

**Review by malwilley — APPROVED** [review]

**Pull request merged**